src/metabase/api/database.clj

```diff
@@ -240,7 +241,8 @@
     (-> (hydrate db (case include
                       "tables"        :tables
                       "tables.fields" [:tables [:fields [:target :has_field_values] :has_field_values]]))
-        (update :tables (partial filter mi/can-read?)))))
+        (update :tables (fn [tables]
+                          (filter #(and (nil? (:visibility_type %)) (mi/can-read? %)) tables))))))
```

Comment:
This isn't a request to change your code, just some feedback for writing more concise code for future reference. 

Instead of `(nil? (:visibility_type %))` you could do `(complement :visibility_type)`. `(:visibility_type table)` is truthy for any value besides `nil` and `false`. So the `complement` would be truthy for `nil` *or* `false`. Since visibility type is a keyword it's never going to be `false`.

You could tighten this up a little further with `every-pred` which combines several predicate functions into a single predicate function as if by `and`:

```clj
(fn [tables]
  (filter (every-pred (complement :visibility_type) mi/can-read?) tables))
```

At that point I'd take it a step further and use `partial` e.g.

```clj
(partial filter (every-pred (complement :visibility_type) mi/can-read?))
```

Now that is definitely a bit more concise. But I think the jury is out on whether it's clearer. Either way the code is fine as it is

Reply:
🎉 thanks for all the tips! They're super helpful as I wade into Clojure's endless supply of small utility functions.